mono/mini/interp/interp.c

```diff
@@ -2624,6 +2627,19 @@ ves_exec_method_with_context (MonoInvocation *frame, ThreadContext *context, uns
 
 			mono_pop_lmf (&ext.lmf);
 
+			if (context->has_resume_state) {
+				/*
+				 * If this bit is set, it means the call has thrown the exception, and we
+				 * reached this point because the EH code in mono_handle_exception ()
+				 * unwound all the JITted frames below us. mono_interp_set_resume_state ()
+				 * has set the fields in context to indicate where we have to resume execution.
+				 */
+				if (frame == context->handler_frame)
+					SET_RESUME_STATE (context);
+				else
+					goto exit_frame;
+			}
```

Comment:
does the icall trampoline also need a check for `has_resume_state` upon returning? (after popping the LMF)